### PR TITLE
MUL-384: BTC Transaction sources from multiple addresses.

### PR DESCRIPTION
**Pull request by Enmk**
BTC transaction now supports inputs from multiple addresses signed with multiple keys.

### CODE REVIEW

File: multy_transaction/properties.h [hunk 1]
```diff
@@ -36,10 +36,9 @@ MULTY_TRANSACTION_API struct Error* properties_set_binary_data_value(
         const char* name,
         const struct BinaryData* value);
 
-MULTY_TRANSACTION_API struct Error* properties_set_public_key_value(
-        struct Properties* properties,
+MULTY_TRANSACTION_API struct Error* properties_set_private_key_value(struct Properties* properties,
         const char* name,
-        const struct PublicKey* value);
+        const PrivateKey* value);
```
Comment (Enmk) [inline multy_transaction/properties.h]: `struct` is missing

File: multy_transaction/internal/properties.cpp [hunk 2]
```diff
@@ -84,8 +83,8 @@ std::string get_value_type_string(ValueType type)
             return "Amount";
         case VALUE_TYPE_BINARY_DATA:
             return "BinaryData";
-        case VALUE_TYPE_PUBLIC_KEY:
-            return "PublicKey";
+        case VALUE_TYPE_PRIVATE_KEY:
+            return "PrivateKey";
             //        case Property::VALUE_TYPE_PRIVATE_KEY:
```
Comment (PashaKlybik) [inline multy_transaction/internal/properties.cpp]: delete this comment

File: multy_transaction/properties.h [hunk 3]
```diff
@@ -36,10 +36,9 @@ MULTY_TRANSACTION_API struct Error* properties_set_binary_data_value(
         const char* name,
         const struct BinaryData* value);
 
-MULTY_TRANSACTION_API struct Error* properties_set_public_key_value(
-        struct Properties* properties,
+MULTY_TRANSACTION_API struct Error* properties_set_private_key_value(struct Properties* properties,
```
Comment (PashaKlybik) [inline multy_transaction/properties.h]: `struct Properties* properties,` to new line

File: multy_transaction/internal/bitcoin_transaction.cpp [hunk 4]
```diff
@@ -636,12 +638,13 @@ void BitcoinTransaction::sign()
...
+            BinaryDataPtr signature = private_key->sign(hash_stream.get_content());
 
             BitcoinDataStream sig_script_stream;
-            sig_script_stream << as_compact_size(sign->len + 1);
-            sig_script_stream << *sign;
+            sig_script_stream << as_compact_size(signature->len + 1);
+            sig_script_stream << *signature;
             sig_script_stream << uint8_t(1); // hash-code type
 
             const BinaryData& public_key
```
Comment (PashaKlybik) [inline multy_transaction/internal/bitcoin_transaction.cpp]: set public_key to this Txin

**Comment by Enmk** [comment]
Fixed

**Pull request merged**